{"cells": [{"metadata": {"id": "Gs8blfz0yc-r"}, "cell_type": "markdown", "source": "**Installing the Required Package**"}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "_0GzR_oJydmT", "outputId": "49d8acd3-9ca8-434e-d675-b47423ecf10c"}, "cell_type": "code", "source": "pip install tensorflow", "execution_count": 7, "outputs": [{"output_type": "stream", "text": "Requirement already satisfied: tensorflow in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (2.7.2)\nRequirement already satisfied: protobuf>=3.9.2 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (3.19.1)\nRequirement already satisfied: h5py>=2.9.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (3.2.1)\nRequirement already satisfied: keras<2.8,>=2.7.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (2.7.0)\nRequirement already satisfied: opt-einsum>=2.3.2 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (3.3.0)\nRequirement already satisfied: astunparse>=1.6.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.6.3)\nRequirement already satisfied: keras-preprocessing>=1.1.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.1.2)\nRequirement already satisfied: gast<0.5.0,>=0.2.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (0.4.0)\nRequirement already satisfied: typing-extensions>=3.6.6 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (4.1.1)\nRequirement already satisfied: tensorflow-estimator<2.8,~=2.7.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (2.7.0)\nRequirement already satisfied: absl-py>=0.4.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (0.12.0)\nRequirement already satisfied: google-pasta>=0.1.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (0.2.0)\nRequirement already satisfied: six>=1.12.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.15.0)\nRequirement already satisfied: tensorboard~=2.7 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (2.7.0)\nRequirement already satisfied: numpy>=1.14.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.20.3)\nRequirement already satisfied: wrapt>=1.11.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.12.1)\nRequirement already satisfied: grpcio<2.0,>=1.24.3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.42.0)\nRequirement already satisfied: termcolor>=1.1.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.1.0)\nRequirement already satisfied: tensorflow-io-gcs-filesystem>=0.21.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (0.23.1)\nRequirement already satisfied: flatbuffers<3.0,>=1.12 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (2.0)\nRequirement already satisfied: wheel<1.0,>=0.32.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (0.37.0)\nRequirement already satisfied: google-auth<3,>=1.6.3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (1.23.0)\nRequirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (0.4.4)\nRequirement already satisfied: markdown>=2.6.8 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (3.3.3)\nRequirement already satisfied: tensorboard-plugin-wit>=1.6.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (1.6.0)\nRequirement already satisfied: requests<3,>=2.21.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (2.26.0)\nRequirement already satisfied: tensorboard-data-server<0.7.0,>=0.6.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (0.6.1)\nRequirement already satisfied: setuptools>=41.0.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (58.0.4)\nRequirement already satisfied: werkzeug>=0.11.15 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (2.0.2)\nRequirement already satisfied: rsa<5,>=3.1.4 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from google-auth<3,>=1.6.3->tensorboard~=2.7->tensorflow) (4.7.2)\nRequirement already satisfied: cachetools<5.0,>=2.0.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from google-auth<3,>=1.6.3->tensorboard~=2.7->tensorflow) (4.2.2)\nRequirement already satisfied: pyasn1-modules>=0.2.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from google-auth<3,>=1.6.3->tensorboard~=2.7->tensorflow) (0.2.8)\nRequirement already satisfied: requests-oauthlib>=0.7.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard~=2.7->tensorflow) (1.3.0)\nRequirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard~=2.7->tensorflow) (0.4.8)\nRequirement already satisfied: charset-normalizer~=2.0.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.7->tensorflow) (2.0.4)\nRequirement already satisfied: certifi>=2017.4.17 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.7->tensorflow) (2022.9.24)\nRequirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.7->tensorflow) (1.26.7)\nRequirement already satisfied: idna<4,>=2.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.7->tensorflow) (3.3)\nRequirement already satisfied: oauthlib>=3.0.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard~=2.7->tensorflow) (3.2.1)\nNote: you may need to restart the kernel to use updated packages.\n", "name": "stdout"}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "V_Pji4co0_Yf", "outputId": "3565d1ea-a04e-4e66-d6b3-499998e6de7f"}, "cell_type": "code", "source": "pip install keras==2.2.4", "execution_count": 8, "outputs": [{"output_type": "stream", "text": "Collecting keras==2.2.4\n  Downloading Keras-2.2.4-py2.py3-none-any.whl (312 kB)\n\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 312 kB 22.8 MB/s eta 0:00:01\n\u001b[?25hRequirement already satisfied: six>=1.9.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (1.15.0)\nRequirement already satisfied: numpy>=1.9.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (1.20.3)\nRequirement already satisfied: scipy>=0.14 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (1.7.3)\nRequirement already satisfied: h5py in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (3.2.1)\nCollecting keras-applications>=1.0.6\n  Downloading Keras_Applications-1.0.8-py3-none-any.whl (50 kB)\n\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 50 kB 19.8 MB/s eta 0:00:01\n\u001b[?25hRequirement already satisfied: keras-preprocessing>=1.0.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (1.1.2)\nRequirement already satisfied: pyyaml in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from keras==2.2.4) (5.4.1)\nInstalling collected packages: keras-applications, keras\n  Attempting uninstall: keras\n    Found existing installation: keras 2.7.0\n    Uninstalling keras-2.7.0:\n      Successfully uninstalled keras-2.7.0\n\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\ntensorflow 2.7.2 requires keras<2.8,>=2.7.0, but you have keras 2.2.4 which is incompatible.\u001b[0m\nSuccessfully installed keras-2.2.4 keras-applications-1.0.8\nNote: you may need to restart the kernel to use updated packages.\n", "name": "stdout"}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "8EqI6gkG1Aqn", "outputId": "7f4e26f4-3f0b-4784-e1f0-9df925901b68"}, "cell_type": "code", "source": "pip install opencv-python", "execution_count": 9, "outputs": [{"output_type": "stream", "text": "Requirement already satisfied: opencv-python in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (4.6.0.66)\nRequirement already satisfied: numpy>=1.17.3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from opencv-python) (1.20.3)\nNote: you may need to restart the kernel to use updated packages.\n", "name": "stdout"}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "6JB5Fw441B3n", "outputId": "65c7f020-5401-4347-af7e-6a431dd9a415"}, "cell_type": "code", "source": "pip install imutils", "execution_count": 10, "outputs": [{"output_type": "stream", "text": "Collecting imutils\n  Downloading imutils-0.5.4.tar.gz (17 kB)\nBuilding wheels for collected packages: imutils\n  Building wheel for imutils (setup.py) ... \u001b[?25ldone\n\u001b[?25h  Created wheel for imutils: filename=imutils-0.5.4-py3-none-any.whl size=25860 sha256=c18e0e234a701910ff319e983a8de0743af78502456d6ad75bbaf71777d7f578\n  Stored in directory: /tmp/wsuser/.cache/pip/wheels/4b/a5/2d/4a070a801d3a3d93f033d3ee9728f470f514826e89952df3ea\nSuccessfully built imutils\nInstalling collected packages: imutils\nSuccessfully installed imutils-0.5.4\nNote: you may need to restart the kernel to use updated packages.\n", "name": "stdout"}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "10nNpuXK1DCf", "outputId": "9fbedc87-f649-4251-e7a1-ddb7f46cf883"}, "cell_type": "code", "source": "pip install flask", "execution_count": 11, "outputs": [{"output_type": "stream", "text": "Collecting flask\n  Downloading Flask-2.2.2-py3-none-any.whl (101 kB)\n\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 101 kB 16.2 MB/s ta 0:00:01\n\u001b[?25hRequirement already satisfied: importlib-metadata>=3.6.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from flask) (4.8.2)\nCollecting Werkzeug>=2.2.2\n  Downloading Werkzeug-2.2.2-py3-none-any.whl (232 kB)\n\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 232 kB 75.8 MB/s eta 0:00:01\n\u001b[?25hRequirement already satisfied: Jinja2>=3.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from flask) (3.0.2)\nCollecting itsdangerous>=2.0\n  Downloading itsdangerous-2.1.2-py3-none-any.whl (15 kB)\nRequirement already satisfied: click>=8.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from flask) (8.0.4)\nRequirement already satisfied: zipp>=0.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from importlib-metadata>=3.6.0->flask) (3.6.0)\nRequirement already satisfied: MarkupSafe>=2.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from Jinja2>=3.0->flask) (2.0.1)\nCollecting MarkupSafe>=2.0\n  Downloading MarkupSafe-2.1.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)\nInstalling collected packages: MarkupSafe, Werkzeug, itsdangerous, flask\n  Attempting uninstall: MarkupSafe\n    Found existing installation: MarkupSafe 2.0.1\n    Uninstalling MarkupSafe-2.0.1:\n      Successfully uninstalled MarkupSafe-2.0.1\n  Attempting uninstall: Werkzeug\n    Found existing installation: Werkzeug 2.0.2\n    Uninstalling Werkzeug-2.0.2:\n      Successfully uninstalled Werkzeug-2.0.2\nSuccessfully installed MarkupSafe-2.1.1 Werkzeug-2.2.2 flask-2.2.2 itsdangerous-2.1.2\nNote: you may need to restart the kernel to use updated packages.\n", "name": "stdout"}]}, {"metadata": {"id": "b47w7hP4y81D"}, "cell_type": "markdown", "source": "**Import ImageDataGenerator Library And Configure It**"}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "CYjey9iuzx4D", "outputId": "f31362cc-2be5-42b2-ee09-11da86ae795c"}, "cell_type": "code", "source": "#Importing the required Libiaries\nfrom tensorflow.keras.preprocessing.image import ImageDataGenerator\nimport tensorflow as tf\nimport os\nfrom tensorflow.keras.models import Sequential\nfrom tensorflow.keras.layers import Dense, Conv2D, Flatten, Dropout, MaxPooling2D\nfrom tensorflow.keras.preprocessing.image import ImageDataGenerator\nimport numpy as np\nimport matplotlib.pyplot as plt\nimport IPython.display as display\nfrom PIL import Image\nimport pathlib", "execution_count": 12, "outputs": []}, {"metadata": {"id": "jSwSD9VSy_5t"}, "cell_type": "code", "source": "from keras.preprocessing.image import ImageDataGenerator\ntrain_datagen = ImageDataGenerator(rescale=1./255, shear_range=0.2, zoom_range=0.2, horizontal_flip=True)\ntest_datagen = ImageDataGenerator(rescale=1./255)", "execution_count": 13, "outputs": []}, {"metadata": {"id": "hfVa8PVSzF6j"}, "cell_type": "markdown", "source": "**Apply ImageDataGenerator Functionality To Train And Test Set**"}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "vXAGjoN20Ll0", "outputId": "0199123a-e6eb-4bb6-da00-ab8edb514497"}, "cell_type": "code", "source": "from tensorflow.keras.preprocessing.image import ImageDataGenerator\nprint(\"This dataset has been created and uploaded by IBM-TeamID-IBM-Project-17580-1659673617\")", "execution_count": 14, "outputs": [{"output_type": "stream", "text": "This dataset has been created and uploaded by IBM-TeamID-IBM-Project-17580-1659673617\n", "name": "stdout"}]}, {"metadata": {"id": "i5f0VSdp0MNU"}, "cell_type": "code", "source": "train_datagen = ImageDataGenerator(rescale=1./255,zoom_range=0.2,horizontal_flip=True, vertical_flip=False)", "execution_count": 15, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "test_datagen= ImageDataGenerator(rescale=1./255)", "execution_count": 16, "outputs": []}, {"metadata": {"id": "2EJ3UxiF0ifN"}, "cell_type": "code", "source": "import os, types\nimport pandas as pd\nfrom botocore.client import Config\nimport ibm_boto3\n\ndef __iter__(self): return 0\n\n# @hidden_cell\n# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.\n# You might want to remove those credentials before you share the notebook.\ncos_client = ibm_boto3.client(service_name='s3',\n    ibm_api_key_id='_8VRpQjGr9MpWRJ4bs6Vc3pefcAhh2FkfcvjpBI2iYXc',\n    ibm_auth_endpoint=\"https://iam.cloud.ibm.com/oidc/token\",\n    config=Config(signature_version='oauth'),\n    endpoint_url='https://s3.private.us.cloud-object-storage.appdomain.cloud')\n\nbucket = 'realtimecommunicationsystempowere-donotdelete-pr-l198s4cogmapus'\nobject_key = 'Dataset.zip'\n\nstreaming_body_1 = cos_client.get_object(Bucket=bucket, Key=object_key)['Body']\n\n# Your data file was loaded into a botocore.response.StreamingBody object.\n# Please read the documentation of ibm_boto3 and pandas to learn more about the possibilities to load the data.\n# ibm_boto3 documentation: https://ibm.github.io/ibm-cos-sdk-python/\n# pandas documentation: http://pandas.pydata.org/\n", "execution_count": 17, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "from io import BytesIO\nimport zipfile\nunzip = zipfile.ZipFile(BytesIO(streaming_body_1.read()),'r')\nfile_paths = unzip.namelist()\nfor path in file_paths:\n    unzip.extract(path)", "execution_count": 18, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "pwd", "execution_count": 19, "outputs": [{"output_type": "execute_result", "execution_count": 19, "data": {"text/plain": "'/home/wsuser/work'"}, "metadata": {}}]}, {"metadata": {}, "cell_type": "code", "source": "\n# @hidden_cell\n# The following code contains the credentials for a file in your IBM Cloud Object Storage.\n# You might want to remove those credentials before you share your notebook.\nmetadata_1 = {\n    'IAM_SERVICE_ID': 'iam-ServiceId-c329fd06-4581-4103-a866-21b0d47afef2',\n    'IBM_API_KEY_ID': '_8VRpQjGr9MpWRJ4bs6Vc3pefcAhh2FkfcvjpBI2iYXc',\n    'ENDPOINT': 'https://s3.private.us.cloud-object-storage.appdomain.cloud',\n    'IBM_AUTH_ENDPOINT': 'https://iam.cloud.ibm.com/oidc/token',\n    'BUCKET': 'realtimecommunicationsystempowere-donotdelete-pr-l198s4cogmapus',\n    'FILE': 'Dataset.zip'\n}\n", "execution_count": 20, "outputs": []}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "_fV80GLL0r6M", "outputId": "f14bee36-66ab-438c-dddf-e033f227fb53"}, "cell_type": "code", "source": "x_train = train_datagen.flow_from_directory('/home/wsuser/work/Dataset/training_set',target_size=(64,64), batch_size=300,\n                                          class_mode='categorical', color_mode = \"grayscale\")", "execution_count": 21, "outputs": [{"output_type": "stream", "text": "Found 15750 images belonging to 9 classes.\n", "name": "stdout"}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "Chf6_b1m0rsd", "outputId": "036be137-00dc-4f30-a5c5-7b948dfd36be"}, "cell_type": "code", "source": "x_test = test_datagen.flow_from_directory('/home/wsuser/work/Dataset/test_set',target_size=(64,64), batch_size=300,\n                                          class_mode='categorical', color_mode = \"grayscale\")", "execution_count": 22, "outputs": [{"output_type": "stream", "text": "Found 2250 images belonging to 9 classes.\n", "name": "stdout"}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "cBDq3PlC0o57", "outputId": "31a7a5e2-c172-446f-b8ce-32e9523cb7db"}, "cell_type": "code", "source": "x_train.class_indices", "execution_count": 23, "outputs": [{"output_type": "execute_result", "execution_count": 23, "data": {"text/plain": "{'A': 0, 'B': 1, 'C': 2, 'D': 3, 'E': 4, 'F': 5, 'G': 6, 'H': 7, 'I': 8}"}, "metadata": {}}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "eu1fxnhY0qUs", "outputId": "49f6ea90-0711-467a-81f7-a72f33b9cbfd"}, "cell_type": "code", "source": "x_test.class_indices", "execution_count": 24, "outputs": [{"output_type": "execute_result", "execution_count": 24, "data": {"text/plain": "{'A': 0, 'B': 1, 'C': 2, 'D': 3, 'E': 4, 'F': 5, 'G': 6, 'H': 7, 'I': 8}"}, "metadata": {}}]}, {"metadata": {"id": "Eo3_A-cAxk3b"}, "cell_type": "markdown", "source": "#**Model Building**"}, {"metadata": {"id": "K66ZuyJqxu_j"}, "cell_type": "markdown", "source": "***Import The Required Model Building Libraries***"}, {"metadata": {"id": "SXMR9HEQx4HW", "colab": {"base_uri": "https://localhost:8080/"}, "outputId": "1256c6b2-f886-4d8b-8919-32591004c1a1"}, "cell_type": "code", "source": "from tensorflow.keras.models import Sequential\nfrom tensorflow.keras.layers import Dense\nfrom tensorflow.keras.layers import Convolution2D\nfrom tensorflow.keras.layers import MaxPooling2D\nfrom tensorflow.keras.layers import Dropout\nfrom tensorflow.keras.layers import Flatten", "execution_count": 25, "outputs": []}, {"metadata": {"id": "TJYdHU9_2SaV"}, "cell_type": "markdown", "source": "**Initialize The Model**"}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "Lvv-H5vr2XmF", "outputId": "b4658267-3d92-4151-99c5-20ac49a4d997"}, "cell_type": "code", "source": "model = Sequential()", "execution_count": 26, "outputs": []}, {"metadata": {"id": "DzjQ0DSz2Z01"}, "cell_type": "markdown", "source": "**Add The Convolution Layer**"}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "ieBnYVYg2fat", "outputId": "c9037864-5cd6-4776-9424-8d59d1680b3f"}, "cell_type": "code", "source": "model.add(Convolution2D(32,(3,3), input_shape=(64,64,1), activation='relu'))", "execution_count": 27, "outputs": []}, {"metadata": {"id": "M49tudUJ2mMM"}, "cell_type": "markdown", "source": "**Add The Pooling Layer**"}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "1vJA-L_D2slU", "outputId": "f0e565c8-d232-4807-955c-ea9b811f5817"}, "cell_type": "code", "source": "model.add(MaxPooling2D(pool_size=(2,2)))", "execution_count": 28, "outputs": []}, {"metadata": {"id": "ryMTPsAj3L1l"}, "cell_type": "markdown", "source": "**Add The Flatten Layer**"}, {"metadata": {"id": "rdtnxWUz3Ny1"}, "cell_type": "code", "source": "model.add(Flatten())", "execution_count": 29, "outputs": []}, {"metadata": {"id": "CxD4N3uF3QhU"}, "cell_type": "markdown", "source": "**Adding The Dense Layers**"}, {"metadata": {"id": "TmcLz9D43UBg"}, "cell_type": "code", "source": "model.add(Dense(units=512, activation='relu'))", "execution_count": 30, "outputs": []}, {"metadata": {"id": "aaOjZg0s3aCt"}, "cell_type": "code", "source": "model.add(Dense(units=9, activation ='softmax'))", "execution_count": 31, "outputs": []}, {"metadata": {"id": "3hXO4Fcr3e4C"}, "cell_type": "markdown", "source": "**Compile The Model**"}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "qe8h7RBL3gjl", "outputId": "f3a783f3-fd59-4a77-e010-23a4844c46ec"}, "cell_type": "code", "source": "model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])", "execution_count": 32, "outputs": []}, {"metadata": {"id": "ErwBVbXP3l2N"}, "cell_type": "markdown", "source": "**Fit And Save The Model**"}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "b2Gfp1d93nXN", "outputId": "125c58be-f738-4384-9dd3-91ad0ff5613c"}, "cell_type": "code", "source": "model.fit_generator(x_train,steps_per_epoch=64, epochs=10, validation_data=x_test, validation_steps=40)", "execution_count": 33, "outputs": [{"output_type": "stream", "text": "/opt/conda/envs/Python-3.9/lib/python3.9/site-packages/tensorflow/python/keras/engine/training.py:1963: UserWarning: `Model.fit_generator` is deprecated and will be removed in a future version. Please use `Model.fit`, which supports generators.\n  warnings.warn('`Model.fit_generator` is deprecated and '\n", "name": "stderr"}, {"output_type": "stream", "text": "Epoch 1/10\n53/64 [=======================>......] - ETA: 10s - loss: 0.7463 - accuracy: 0.7959WARNING:tensorflow:Your input ran out of data; interrupting training. Make sure that your dataset or generator can generate at least `steps_per_epoch * epochs` batches (in this case, 640 batches). You may need to use the repeat() function when building your dataset.\nWARNING:tensorflow:Your input ran out of data; interrupting training. Make sure that your dataset or generator can generate at least `steps_per_epoch * epochs` batches (in this case, 40 batches). You may need to use the repeat() function when building your dataset.\n64/64 [==============================] - 54s 836ms/step - loss: 0.7463 - accuracy: 0.7959 - val_loss: 0.2946 - val_accuracy: 0.9293\n", "name": "stdout"}, {"output_type": "execute_result", "execution_count": 33, "data": {"text/plain": "<tensorflow.python.keras.callbacks.History at 0x7f883b0f56a0>"}, "metadata": {}}]}, {"metadata": {"id": "mArG16SQ3qeF"}, "cell_type": "code", "source": "model.save('aslpng1.h5')", "execution_count": 34, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "!tar -zcvf image-classification-model_new.tgz aslpng1.h5", "execution_count": 35, "outputs": [{"output_type": "stream", "text": "aslpng1.h5\r\n", "name": "stdout"}]}, {"metadata": {}, "cell_type": "code", "source": "ls", "execution_count": 36, "outputs": [{"output_type": "stream", "text": "aslpng1.h5  \u001b[0m\u001b[01;34mDataset\u001b[0m/  image-classification-model_new.tgz\r\n", "name": "stdout"}]}, {"metadata": {}, "cell_type": "code", "source": "!pip install watson-machine-learning-client --upgrade", "execution_count": 37, "outputs": [{"output_type": "stream", "text": "Collecting watson-machine-learning-client\n  Downloading watson_machine_learning_client-1.0.391-py3-none-any.whl (538 kB)\n\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 538 kB 19.7 MB/s eta 0:00:01\n\u001b[?25hRequirement already satisfied: lomond in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (0.3.3)\nRequirement already satisfied: boto3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (1.18.21)\nRequirement already satisfied: urllib3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (1.26.7)\nRequirement already satisfied: tqdm in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (4.62.3)\nRequirement already satisfied: pandas in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (1.3.4)\nRequirement already satisfied: tabulate in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (0.8.9)\nRequirement already satisfied: requests in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (2.26.0)\nRequirement already satisfied: ibm-cos-sdk in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (2.11.0)\nRequirement already satisfied: certifi in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (2022.9.24)\nRequirement already satisfied: botocore<1.22.0,>=1.21.21 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from boto3->watson-machine-learning-client) (1.21.41)\nRequirement already satisfied: jmespath<1.0.0,>=0.7.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from boto3->watson-machine-learning-client) (0.10.0)\nRequirement already satisfied: s3transfer<0.6.0,>=0.5.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from boto3->watson-machine-learning-client) (0.5.0)\nRequirement already satisfied: python-dateutil<3.0.0,>=2.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from botocore<1.22.0,>=1.21.21->boto3->watson-machine-learning-client) (2.8.2)\nRequirement already satisfied: six>=1.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from python-dateutil<3.0.0,>=2.1->botocore<1.22.0,>=1.21.21->boto3->watson-machine-learning-client) (1.15.0)\nRequirement already satisfied: ibm-cos-sdk-core==2.11.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from ibm-cos-sdk->watson-machine-learning-client) (2.11.0)\nRequirement already satisfied: ibm-cos-sdk-s3transfer==2.11.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from ibm-cos-sdk->watson-machine-learning-client) (2.11.0)\nRequirement already satisfied: idna<4,>=2.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests->watson-machine-learning-client) (3.3)\nRequirement already satisfied: charset-normalizer~=2.0.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests->watson-machine-learning-client) (2.0.4)\nRequirement already satisfied: pytz>=2017.3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from pandas->watson-machine-learning-client) (2021.3)\nRequirement already satisfied: numpy>=1.17.3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from pandas->watson-machine-learning-client) (1.20.3)\nInstalling collected packages: watson-machine-learning-client\nSuccessfully installed watson-machine-learning-client-1.0.391\n", "name": "stdout"}]}, {"metadata": {}, "cell_type": "code", "source": "from ibm_watson_machine_learning import APIClient\nwml_credentials = {\n    \"url\" : \"https://us-south.ml.cloud.ibm.com\",\n    \"apikey\":\"JyBAO2rWspFkmBPCn7OvFUzGcv-ufLkzsL4BRzyqZpb6\"\n}\nclient = APIClient(wml_credentials)", "execution_count": 38, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "client = APIClient(wml_credentials)", "execution_count": 39, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "def guid_from_space_name(client, space_name):\n    space = client.spaces.get_details()\n    #print(space)\n    return(next(item for item in space['resources'] if item['entity'][\"name\"] == space_name)['metadata']['id'])", "execution_count": 40, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "space_uid = guid_from_space_name(client, 'Real-Time Communication System Powered by AI for Specially Abled Deployment')\nprint(\"Space UID - \" + space_uid)", "execution_count": 41, "outputs": [{"output_type": "stream", "text": "Space UID - e05e4915-4101-4b09-8512-189181d2f773\n", "name": "stdout"}]}, {"metadata": {}, "cell_type": "code", "source": "client.set.default_space(space_uid)", "execution_count": 42, "outputs": [{"output_type": "execute_result", "execution_count": 42, "data": {"text/plain": "'SUCCESS'"}, "metadata": {}}]}, {"metadata": {}, "cell_type": "code", "source": "client.software_specifications.list()", "execution_count": 43, "outputs": [{"output_type": "stream", "text": "-----------------------------  ------------------------------------  ----\nNAME                           ASSET_ID                              TYPE\ndefault_py3.6                  0062b8c9-8b7d-44a0-a9b9-46c416adcbd9  base\nkernel-spark3.2-scala2.12      020d69ce-7ac1-5e68-ac1a-31189867356a  base\npytorch-onnx_1.3-py3.7-edt     069ea134-3346-5748-b513-49120e15d288  base\nscikit-learn_0.20-py3.6        09c5a1d0-9c1e-4473-a344-eb7b665ff687  base\nspark-mllib_3.0-scala_2.12     09f4cff0-90a7-5899-b9ed-1ef348aebdee  base\npytorch-onnx_rt22.1-py3.9      0b848dd4-e681-5599-be41-b5f6fccc6471  base\nai-function_0.1-py3.6          0cdb0f1e-5376-4f4d-92dd-da3b69aa9bda  base\nshiny-r3.6                     0e6e79df-875e-4f24-8ae9-62dcc2148306  base\ntensorflow_2.4-py3.7-horovod   1092590a-307d-563d-9b62-4eb7d64b3f22  base\npytorch_1.1-py3.6              10ac12d6-6b30-4ccd-8392-3e922c096a92  base\ntensorflow_1.15-py3.6-ddl      111e41b3-de2d-5422-a4d6-bf776828c4b7  base\nautoai-kb_rt22.2-py3.10        125b6d9a-5b1f-5e8d-972a-b251688ccf40  base\nruntime-22.1-py3.9             12b83a17-24d8-5082-900f-0ab31fbfd3cb  base\nscikit-learn_0.22-py3.6        154010fa-5b3b-4ac1-82af-4d5ee5abbc85  base\ndefault_r3.6                   1b70aec3-ab34-4b87-8aa0-a4a3c8296a36  base\npytorch-onnx_1.3-py3.6         1bc6029a-cc97-56da-b8e0-39c3880dbbe7  base\nkernel-spark3.3-r3.6           1c9e5454-f216-59dd-a20e-474a5cdf5988  base\npytorch-onnx_rt22.1-py3.9-edt  1d362186-7ad5-5b59-8b6c-9d0880bde37f  base\ntensorflow_2.1-py3.6           1eb25b84-d6ed-5dde-b6a5-3fbdf1665666  base\nspark-mllib_3.2                20047f72-0a98-58c7-9ff5-a77b012eb8f5  base\ntensorflow_2.4-py3.8-horovod   217c16f6-178f-56bf-824a-b19f20564c49  base\nruntime-22.1-py3.9-cuda        26215f05-08c3-5a41-a1b0-da66306ce658  base\ndo_py3.8                       295addb5-9ef9-547e-9bf4-92ae3563e720  base\nautoai-ts_3.8-py3.8            2aa0c932-798f-5ae9-abd6-15e0c2402fb5  base\ntensorflow_1.15-py3.6          2b73a275-7cbf-420b-a912-eae7f436e0bc  base\nkernel-spark3.3-py3.9          2b7961e2-e3b1-5a8c-a491-482c8368839a  base\npytorch_1.2-py3.6              2c8ef57d-2687-4b7d-acce-01f94976dac1  base\nspark-mllib_2.3                2e51f700-bca0-4b0d-88dc-5c6791338875  base\npytorch-onnx_1.1-py3.6-edt     32983cea-3f32-4400-8965-dde874a8d67e  base\nspark-mllib_3.0-py37           36507ebe-8770-55ba-ab2a-eafe787600e9  base\nspark-mllib_2.4                390d21f8-e58b-4fac-9c55-d7ceda621326  base\nautoai-ts_rt22.2-py3.10        396b2e83-0953-5b86-9a55-7ce1628a406f  base\nxgboost_0.82-py3.6             39e31acd-5f30-41dc-ae44-60233c80306e  base\npytorch-onnx_1.2-py3.6-edt     40589d0e-7019-4e28-8daa-fb03b6f4fe12  base\npytorch-onnx_rt22.2-py3.10     40e73f55-783a-5535-b3fa-0c8b94291431  base\ndefault_r36py38                41c247d3-45f8-5a71-b065-8580229facf0  base\nautoai-ts_rt22.1-py3.9         4269d26e-07ba-5d40-8f66-2d495b0c71f7  base\nautoai-obm_3.0                 42b92e18-d9ab-567f-988a-4240ba1ed5f7  base\npmml-3.0_4.3                   493bcb95-16f1-5bc5-bee8-81b8af80e9c7  base\nspark-mllib_2.4-r_3.6          49403dff-92e9-4c87-a3d7-a42d0021c095  base\nxgboost_0.90-py3.6             4ff8d6c2-1343-4c18-85e1-689c965304d3  base\npytorch-onnx_1.1-py3.6         50f95b2a-bc16-43bb-bc94-b0bed208c60b  base\nautoai-ts_3.9-py3.8            52c57136-80fa-572e-8728-a5e7cbb42cde  base\nspark-mllib_2.4-scala_2.11     55a70f99-7320-4be5-9fb9-9edb5a443af5  base\nspark-mllib_3.0                5c1b0ca2-4977-5c2e-9439-ffd44ea8ffe9  base\nautoai-obm_2.0                 5c2e37fa-80b8-5e77-840f-d912469614ee  base\nspss-modeler_18.1              5c3cad7e-507f-4b2a-a9a3-ab53a21dee8b  base\ncuda-py3.8                     5d3232bf-c86b-5df4-a2cd-7bb870a1cd4e  base\nautoai-kb_3.1-py3.7            632d4b22-10aa-5180-88f0-f52dfb6444d7  base\npytorch-onnx_1.7-py3.8         634d3cdc-b562-5bf9-a2d4-ea90a478456b  base\n-----------------------------  ------------------------------------  ----\nNote: Only first 50 records were displayed. To display more use 'limit' parameter.\n", "name": "stdout"}]}, {"metadata": {}, "cell_type": "code", "source": "software_spec_uid = client.software_specifications.get_uid_by_name(\"tensorflow_rt22.1-py3.9\")\nsoftware_spec_uid", "execution_count": 44, "outputs": [{"output_type": "execute_result", "execution_count": 44, "data": {"text/plain": "'acd9c798-6974-5d2f-a657-ce06e986df4d'"}, "metadata": {}}]}, {"metadata": {}, "cell_type": "code", "source": "\nmodel_details = client.repository.store_model(model='image-classification-model_new.tgz', meta_props={\n client.repository.ModelMetaNames.NAME:\"CNN_image-classification-model_new\",\n client.repository.ModelMetaNames.TYPE:\"tensorflow_2.7\",\n client.repository.ModelMetaNames.SOFTWARE_SPEC_UID:software_spec_uid\n})\nmodel_details", "execution_count": 45, "outputs": [{"output_type": "execute_result", "execution_count": 45, "data": {"text/plain": "{'entity': {'hybrid_pipeline_software_specs': [],\n  'software_spec': {'id': 'acd9c798-6974-5d2f-a657-ce06e986df4d',\n   'name': 'tensorflow_rt22.1-py3.9'},\n  'type': 'tensorflow_2.7'},\n 'metadata': {'created_at': '2022-11-12T05:08:26.822Z',\n  'id': 'a5e6019b-219e-40a9-881e-310b17f2298b',\n  'modified_at': '2022-11-12T05:08:39.238Z',\n  'name': 'CNN_image-classification-model_new',\n  'owner': 'IBMid-6610045EH0',\n  'resource_key': '4fee2594-d7e9-4f43-aae7-ea71c40b0653',\n  'space_id': 'e05e4915-4101-4b09-8512-189181d2f773'},\n 'system': {'warnings': []}}"}, "metadata": {}}]}, {"metadata": {}, "cell_type": "code", "source": "model_id = client.repository.get_model_id(model_details)\nmodel_id", "execution_count": 46, "outputs": [{"output_type": "execute_result", "execution_count": 46, "data": {"text/plain": "'a5e6019b-219e-40a9-881e-310b17f2298b'"}, "metadata": {}}]}, {"metadata": {}, "cell_type": "code", "source": "client.repository.list()", "execution_count": 47, "outputs": [{"output_type": "stream", "text": "------------------------------------  ----------------------------------  ------------------------  --------------  -----\nGUID                                  NAME                                CREATED                   FRAMEWORK       TYPE\na5e6019b-219e-40a9-881e-310b17f2298b  CNN_image-classification-model_new  2022-11-12T05:08:26.002Z  tensorflow_2.7  model\n757e1ecb-b887-4fd8-bc9b-42a3afa33c72  CNN_image-classification-model_new  2022-11-09T07:37:29.002Z  tensorflow_2.7  model\n------------------------------------  ----------------------------------  ------------------------  --------------  -----\n", "name": "stdout"}]}, {"metadata": {}, "cell_type": "code", "source": "client.repository.download(model_id,'my_model.tar.gz')", "execution_count": 48, "outputs": [{"output_type": "stream", "text": "Successfully saved model content to file: 'my_model.tar.gz'\n", "name": "stdout"}, {"output_type": "execute_result", "execution_count": 48, "data": {"text/plain": "'/home/wsuser/work/my_model.tar.gz'"}, "metadata": {}}]}, {"metadata": {"id": "yiaYYV9iei3z"}, "cell_type": "markdown", "source": "# **Test The Model**"}, {"metadata": {"id": "jkLy2akoep_M"}, "cell_type": "markdown", "source": "**Import The Packages**"}, {"metadata": {}, "cell_type": "code", "source": "pip install numpy", "execution_count": 49, "outputs": [{"output_type": "stream", "text": "Requirement already satisfied: numpy in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (1.20.3)\nNote: you may need to restart the kernel to use updated packages.\n", "name": "stdout"}]}, {"metadata": {}, "cell_type": "code", "source": "pip install opencv-python", "execution_count": 50, "outputs": [{"output_type": "stream", "text": "Requirement already satisfied: opencv-python in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (4.6.0.66)\nRequirement already satisfied: numpy>=1.14.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from opencv-python) (1.20.3)\nNote: you may need to restart the kernel to use updated packages.\n", "name": "stdout"}]}, {"metadata": {"id": "RAqqR4laevDq"}, "cell_type": "code", "source": "from keras.models import load_model\nimport numpy as np\n#import cv2", "execution_count": 68, "outputs": []}, {"metadata": {"id": "DauajNmxe0zc"}, "cell_type": "markdown", "source": "**Load The Saved Model**"}, {"metadata": {}, "cell_type": "code", "source": "ls", "execution_count": 70, "outputs": [{"output_type": "stream", "text": "aslpng1.h5  \u001b[0m\u001b[01;34mDataset\u001b[0m/  image-classification-model_new.tgz  my_model.tar.gz\r\n", "name": "stdout"}]}, {"metadata": {"id": "aubEakBIe3nS"}, "cell_type": "code", "source": "model = load_model(r'aslpng1.h5')", "execution_count": 71, "outputs": [{"output_type": "error", "ename": "ValueError", "evalue": "Unknown layer: Sequential. Please ensure this object is passed to the `custom_objects` argument. See https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details.", "traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)", "\u001b[0;32m/tmp/wsuser/ipykernel_165/148233257.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mmodel\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mload_model\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mr'aslpng1.h5'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m", "\u001b[0;32m/opt/conda/envs/Python-3.9/lib/python3.9/site-packages/keras/utils/traceback_utils.py\u001b[0m in \u001b[0;36merror_handler\u001b[0;34m(*args, **kwargs)\u001b[0m\n", "\u001b[0;32m/opt/conda/envs/Python-3.9/lib/python3.9/site-packages/keras/utils/generic_utils.py\u001b[0m in \u001b[0;36mclass_and_config_for_serialized_keras_object\u001b[0;34m(config, module_objects, custom_objects, printable_module_name)\u001b[0m\n\u001b[1;32m    559\u001b[0m     \u001b[0;31m# Example\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    560\u001b[0m     \u001b[0;31m`\u001b[0m\u001b[0;31m`\u001b[0m\u001b[0;31m`\u001b[0m\u001b[0mpython\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 561\u001b[0;31m         \u001b[0;34m>>\u001b[0m\u001b[0;34m>\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mkeras\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mutils\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgeneric_utils\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mtranspose_shape\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    562\u001b[0m         \u001b[0;34m>>\u001b[0m\u001b[0;34m>\u001b[0m \u001b[0mtranspose_shape\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m16\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m128\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m128\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m32\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'channels_first'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mspatial_axes\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    563\u001b[0m         \u001b[0;34m(\u001b[0m\u001b[0;36m16\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m32\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m128\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m128\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;31mValueError\u001b[0m: Unknown layer: Sequential. Please ensure this object is passed to the `custom_objects` argument. See https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details."]}]}, {"metadata": {"id": "jQjh3WEej2T-"}, "cell_type": "markdown", "source": "**Load The Test Image, Pre-Process It And Predict**"}, {"metadata": {"id": "4BX6QPsxjxOx"}, "cell_type": "code", "source": "from skimage.transform import resize\ndef detect(frame):\n  img = resize(frame,(64,64,1))\n  img = np.expand_dims(img,axis=0)\n  if(np.max(img)>1):\n    img = img/255.0\n  pred = model.predict(img)\n  print(pred)\n  pred1 = model.predict(img)\n  print(pred1)", "execution_count": 72, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "conda install opencv", "execution_count": null, "outputs": [{"output_type": "stream", "text": "Collecting package metadata (current_repodata.json): - ", "name": "stdout"}]}, {"metadata": {"id": "kCkufbDbk5ZS"}, "cell_type": "code", "source": "frame = cv2.imread(r\"/content/drive/MyDrive/Dataset/test_set/G/1.png\")\n", "execution_count": 1, "outputs": [{"output_type": "error", "ename": "NameError", "evalue": "name 'cv2' is not defined", "traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)", "\u001b[0;32m/tmp/wsuser/ipykernel_2159/3676648325.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mframe\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcv2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mimread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mr\"/content/drive/MyDrive/Dataset/test_set/G/1.png\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m", "\u001b[0;31mNameError\u001b[0m: name 'cv2' is not defined"]}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/", "height": 339}, "id": "wJCqhBTrl9TR", "outputId": "7f736a34-1bd8-4360-d7bb-304e332034dd"}, "cell_type": "code", "source": "data = detect(frame)\nprint(data)", "execution_count": null, "outputs": [{"output_type": "stream", "name": "stdout", "text": "1/1 [==============================] - 0s 29ms/step\n[[1.4740882e-13 7.6793548e-16 5.0368483e-09 5.0675267e-14 5.5240919e-13\n  5.8500434e-11 1.0000000e+00 1.2302905e-08 6.8426097e-13]]\n"}, {"output_type": "error", "ename": "AttributeError", "evalue": "ignored", "traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)", "\u001b[0;32m<ipython-input-18-d689e9e9360d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdetect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mframe\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;32m<ipython-input-10-f8a49c3080f0>\u001b[0m in \u001b[0;36mdetect\u001b[0;34m(frame)\u001b[0m\n\u001b[1;32m      7\u001b[0m   \u001b[0mpred\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmodel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpredict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mimg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m   \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpred\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m   \u001b[0mpred1\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmodel\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpredict_classes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mimg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     10\u001b[0m   \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpred1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;31mAttributeError\u001b[0m: 'Sequential' object has no attribute 'predict_classes'"]}]}], "metadata": {"colab": {"provenance": [], "collapsed_sections": []}, "kernelspec": {"name": "python3", "display_name": "Python 3.9", "language": "python"}, "language_info": {"name": "python", "version": "3.9.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}}, "nbformat": 4, "nbformat_minor": 1}